### PR TITLE
ACC and CRPS

### PR DESCRIPTION
**Pull request by dallasfoster**
# Earth2Studio Pull Request

## Description
Adds anomaly correlation coefficient (ACC) and continuous ranked probability score (CRPS) as new metrics.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: earth2studio/statistics/acc.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        Note that these weights must have the same number of dimensions
+        as passed in reduction_dimensions.
+        Example: if reduction_dimensions = ['lat', 'lon'] then
+        assert weights.ndim == 2.
+    """
+
+    def __init__(
+        self,
+        reduction_dimensions: list[str],
+        climatology: DataSource = None,
```
Comment (NickGeneva) [inline earth2studio/statistics/acc.py]: climatology: DataSource | None = None
weights: torch.Tensor | None = None

File: earth2studio/statistics/acc.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            Input tensor #2 intended to apply statistic to.
+        y_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `y` tensor.
+            'reduction_dims' must be in coords.
+
+        Returns
+        -------
+        tuple[torch.Tensor, CoordSystem]
+            Returns anomaly correlation coefficient tensor with appropriate reduced coordinates.
+        """
```
Comment (NickGeneva) [inline earth2studio/statistics/acc.py:104]: Maybe add a handshake_dim(coords, reduction_dim)

To check coords have valid dims here and give informative error if not

File: earth2studio/statistics/acc.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            self.weights, self.reduction_dimensions, x_coords
+        ).to(x.device)
+        weights_sum = torch.sum(weights)
+
+        # Get climatology information
+        if self.climatology is not None:
+            if "lead_time" in output_coords:
+                clim, _ = fetch_data(
+                    self.climatology,
+                    output_coords["time"],
```
Comment (NickGeneva) [inline earth2studio/statistics/acc.py:132]: Does this also imply that the inputs must have time, variable axis? May want to add this to the call doc string.

I guess no data source would not require this. But I think it would be alright to have a requirement for these two...

File: earth2studio/statistics/acc.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        y : torch.Tensor
+            Input tensor #2 intended to apply statistic to.
+        y_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `y` tensor.
+            'reduction_dims' must be in coords.
+
+        Returns
+        -------
+        tuple[torch.Tensor, CoordSystem]
+            Returns anomaly correlation coefficient tensor with appropriate reduced coordinates.
```
Comment (NickGeneva) [inline earth2studio/statistics/acc.py:99]: Similar to Modulus consider adding:

 Note
 ----
 For additional information see reference: 
 
 - https://www.atmos.albany.edu/daes/atmclasses/atm401/spring_2016/ppts_pdfs/ECMWF_ACC_definition.pdf

File: earth2studio/statistics/crps.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                handshake_coords(y_coords, x_coords, c)
+                coord_count += 1
+
+        dim = list(x_coords).index(self.reduction_dimension)
+
+        _crps = _crps_from_empirical_cdf(x, y, dim=dim)
+        return _crps, y_coords.copy()
+
+
+def _crps_from_empirical_cdf(
```
Comment (NickGeneva) [inline earth2studio/statistics/crps.py:116]: Is this to eventually be replaced by the modulus version getting upstreamed?

If so maybe add a Note here to eventually replace this.

File: earth2studio/statistics/crps.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+class crps:
+    """
+    Compute the Continuous Ranked Probably Score (CRPS).
+
+    Uses this formula
+        # int [F(x) - 1(x-y)]^2 dx
+
+    where F is the emperical CDF and 1(x-y) = 1 if x > y.
+
+    This method is more memory efficient than the kernel method, and uses O(n
```
Comment (NickGeneva) [inline earth2studio/statistics/crps.py]: "This method is more memory efficient than the kernel method"

Seems repeated with whats below, is this comment about kernel method needed here?

File: earth2studio/statistics/crps.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+        cdf = (i + 1) / n
+
+        # a. case y < x0
+        val = (x1 - x0) * (cdf - 1) ** 2
+        mask = truth < x0
+        ans += torch.where(mask, val, 0.0)
+
+        # b. case x0 <= y <= x1
+        val = (truth - x0) * cdf**2 + (x1 - truth) * (cdf - 1) ** 2
```
Comment (NickGeneva) [inline earth2studio/statistics/crps.py:177]: Is this linear interpolation better than sticking with the true definition of 1(x-y).

By definition, 1(x-y) means anything greater than x is 0 no?

File: earth2studio/statistics/crps.py (hunk 5 above)
Comment (dallasfoster) [reply earth2studio/statistics/crps.py:116]: Good idea, thanks

**Review by NickGeneva — APPROVED** [review]

File: earth2studio/statistics/crps.py (hunk 7 above)
Comment (dallasfoster) [reply earth2studio/statistics/crps.py:177]: Interesting point
